linalg/solvers.hpp

```diff
@@ -799,6 +800,61 @@ class KLUSolver : public Solver
 
 #endif // MFEM_USE_SUITESPARSE
 
+/// Block diagonal solver for A, each block is inverted by direct solver
+class BlockDiagSolver : public Solver
+{
+   SparseMatrix& block_dof_;
```

Comment:
To go with MFEM style, these member variables should not have trailing `_` in the library (in the miniapp itself it's fine).